common/app/views/support/HtmlCleaner.scala

```diff
@@ -602,3 +603,24 @@ object MembershipEventCleaner extends HtmlCleaner {
     document
   }
 }
+
+object ChaptersLinksCleaner extends HtmlCleaner {
+  def slugify(text: String): String = {
+    Normalizer.normalize(text, Normalizer.Form.NFKD)
+      .toLowerCase
+      .replaceAll("[^0-9a-z ]", "")
+      .trim.replaceAll(" +", "-")
+  }
+
+  override def clean(document: Document): Document = {
+    if(ChapterHeadingsSwitch.isSwitchedOn) {
+      val autoaChapters = document.getElementsByClass("auto-chapter")
+
+      autoaChapters.foreach { ch =>
+        val h2 = ch.getElementsByTag("h2")
+        h2.attr("id", slugify(h2.text())).attr("class", "anchor-link-fix")
```

Comment:
We could apply the fix styles to all `h2`s? See here for more info on the hack: http://stackoverflow.com/questions/11386807/adding-an-automatic-offset-to-the-scroll-position-for-all-hash-links-calls